spec/graphql/types/query_type_spec.rb

```diff
@@ -1,32 +1,132 @@
+          expect(first_arg.type).to be(GraphQL::Types::Int)
+          expect(order_by_arg.type).to be(MadekGraphqlSchema::OrderByEnum)
+        end
+      end
+
+      context 'response' do
+        before(:all) do
+          FactoryGirl.create_list(:media_entry_with_title, 101)
+        end
+
+        context 'for query with no arguments specified' do
+          let(:query) { media_entries_query }
+          let(:response_data) { response_as_hash(query)[:data][:allMediaEntries] }
+          let(:stringified_created_ats) { MediaEntry.order('created_at DESC').
+                                          first(100).
+                                          pluck(:created_at).
+                                          map(&:to_s) }
+
+          it 'returns first 100 MediaEntries ordered by CREATED_AT_DESC' do
+            expect(stringified_created_ats_from_response(response_data)).
+              to eq(stringified_created_ats)
+          end
+        end
+
+        context 'for query with arguments' do
+          let(:query) { media_entries_query(first: 11,
+                                            order_by: 'CREATED_AT_ASC') }
+          let(:response_data) { response_as_hash(query)[:data][:allMediaEntries] }
+          let(:stringified_created_ats) { MediaEntry.order('created_at ASC').
+                                          first(11).
+                                          pluck(:created_at).
+                                          map(&:to_s) }
+
+          it 'returns specified number of media entries in specified order' do
+            expect(stringified_created_ats_from_response(response_data)).
+              to eq(stringified_created_ats)
+          end
+        end
+      end
     end
 
-    it 'defines a field created_at of String type' do
-      expect(subject).to have_field(:createdAt).that_returns(type('ISO8601DateTime'))
+
+    def stringified_created_ats_from_response(response)
+      response.map do |media_entry|
+        ActiveSupport::TimeZone['UTC'].parse(media_entry[:createdAt]).to_s
+      end
     end
 
-    it 'defines a field title of String type' do
-      expect(subject).to have_field(:id).that_returns(type('String'))
+    def media_entry_query(id)
+      <<-GRAPHQL
+        {
+          mediaEntry(id: "#{id}") {
+            id
+            createdAt
+            title
+          }
+        }
+      GRAPHQL
     end
 
+    def media_entries_query(first: nil, order_by: nil)
+      first = "first: #{first}" if first
+      order_by = "orderBy: #{order_by}" if order_by
+      params = [first, order_by].join(', ')
 
+      <<-GRAPHQL
+        {
+          allMediaEntries(#{params}) {
+            id
+            createdAt
+            title
+          }
+        }
+      GRAPHQL
+    end
 
-    def type(type)
-      "GraphQL::Types::#{type}"
+    def response_as_hash(query)
+      MadekGraphqlSchema.execute(query).to_h.with_indifferent_access
```

Comment:
@kooropatfa what's the purpose of calling `.with_indifferent_access` in this case? You used symbol keys only in the code above, so maybe `.deep_symbolize_keys` would be a better choice? Or just leave the string keys as they're returned from response.

Reply:
It's because I prefer calling hash values by symbol than by string because symbol it's faster to type and as readable as string. But you're right that `deep_symbolize_keys` is better choice